pages/content/amp-dev/search.html

```diff
@@ -0,0 +1,155 @@
+---
+$title: Search
+$order: 1
+description: Search amp.dev.
+$view: /views/overview/custom-content.j2
+---
+{% do doc.styles.addCssFile('css/components/templates/custom-content.css') %}
+
+
+{% do doc.styles.addCssFile('/css/components/molecules/biggy.css') %}
+{% do doc.styles.addCssFile('css/components/organisms/teaser-grid.css') %}
+{% do doc.styles.addCssFile('/css/components/molecules/copy.css') %}
+
+{% do doc.amp_dependencies.add('amp-autocomplete', '0.1') %}
+{% do doc.amp_dependencies.add('amp-form', '0.1') %}
+{% do doc.amp_dependencies.add('amp-bind', '0.1') %}
+{% do doc.amp_dependencies.add('amp-list', '0.1') %}
+{% do doc.amp_dependencies.add('amp-mustache', '0.2', 'template') %}
+
+
+
+<section class="ap--container">
+
+  <div class="ap-m-copy ap-m-copy-center-small ap-m-biggy-background">
+    <h1>Search</h1>
+    <form action-xhr="/search/do" class="search-form" id="searchForm" method="GET" on="submit-success: AMP.setState({'query': tempQuery, 'pageNumber': 1, 'showResults': true, 'search': event.response})"
+          target="_top">
+      <amp-autocomplete filter="substring"
+                        min-characters="0"
+                        on="select:AMP.setState({ 'tempQuery': event.value }),searchForm.submit"
+                        submit-on-enter="false"
+                        src="/search/autosuggest"
+      >
+        <input name="q" on="change:AMP.setState({ 'tempQuery': event.value })"
+               placeholder="What are you looking for?"
+               required>
+      </amp-autocomplete>
+      <input name="page" type="hidden" value="1">
+      <input name="locale" type="hidden" value="{{ doc.locale }}">
+      <button type="submit">Search</button>
+    </form>
+    <div [hidden]="showResults != true" hidden>
+      <p [text]="search.result.totalResults + ' results found'" class="info"></p>
+
+      <amp-list [height]="(100) * search.result.components.length + (80) * search.result.pages.length"
+                [src]="'/search/do?q=' + query + '&page=' + pageNumber + '&locale={{ doc.locale }}'"
```

Comment:
Why not bind to `[src]=search`?

Reply:
`load-more-bookmark="nextUrl"` did not work with that :-(

Reply:
ah - too bad. One thing you can do is trigger the amp-list request straight away: `on="submit: AMP.setState({'query': tempQuery, 'pageNumber': 1, 'showResults': true})`. You can then set the action to a dummy endpoint, e.g.: `action-xhr="https://amp.dev/documentation/examples/api/echo"`.

Reply:
I changed the logic, so that the form will display the first page and the amp list will display the rest. Now there is no duplicate request. Although the second page is loaded directly, even when it is not visible. But maybe the problem is, that the amp list is in the viewport when when the results from the first page are not rendered. But I think all in all it is much cleaner now.